app/deepseg.cc

```diff
@@ -615,9 +630,26 @@ int main(int argc, char* argv[]) try {
 
 
 	// Processing components, all at capture true geometry
-	cv::Mat mask(capGeo.value().second, capGeo.value().first, CV_8U);
+    cv::Mat mask;
+    if ( crop_region.height ) {
+		cv::Mat masktmp((int)(crop_region.height), (int)(crop_region.width), CV_8U);
+		mask = masktmp;
+	} else {
+		cv::Mat masktmp(capGeo.value().second, capGeo.value().first, CV_8U);
+		mask = masktmp;
+	}
```

Comment:
Optimize similar to example shown above. `masktmp` is unused (except for assignment). Drop unnecessary parens.

Reply:
See previous not committed suggestion.

Reply:
This may possibly be:
```
	std::pair<int, int> bg_dim = {(int)capGeo.value().first, (int)capGeo.value().second};
	if ( crop_region.height ) {
		bg_dim = {crop_region.width, crop_region.height};
	}
	cv::Mat mask(bg_dim.value().first, bg_dim.value().second, CV_8U);
```

Reply:
Replace `std::pair<size_t, size_t>` by `cv::Size` and the overall code will magically shrink quite a bit …